zebrad/src/components/tracing/component.rs

```diff
@@ -16,14 +22,23 @@ pub struct Tracing {
     /// The installed filter reloading handle, if enabled.
     //
     // TODO: when fmt::Subscriber supports per-layer filtering, remove the Option
-    filter_handle: Option<Handle<EnvFilter, Formatter>>,
+    filter_handle: Option<
+        Handle<
+            EnvFilter,
+            Formatter<format::DefaultFields, format::Format<format::Full>, NonBlocking>,
+        >,
+    >,
 
     /// The originally configured filter.
     initial_filter: String,
 
     /// The installed flame graph collector, if enabled.
     #[cfg(feature = "flamegraph")]
     flamegrapher: Option<flame::Grapher>,
+
+    /// Drop guard for worker thread of non-blocking logger,
+    /// responsible for flushing any remaining logs when the program terminates
+    _guard: WorkerGuard,
```

Comment:
Zebra calls `sys::exit()` in a few different places, how can we make sure the guard is dropped before that happens?

This is the standard shutdown after a Ctrl-C. I think we should have called all the destructors here, but I'm not sure. Do we need a test for that?
https://github.com/ZcashFoundation/zebra/blob/3ff56c22cda4504832d4fe57fd6357e8743965ed/zebrad/src/application.rs#L468-L475

We only call this code during tests, so we've decided not to fix the `exit()` calls here. It's ok if we lose a bit of output in the test, as long as the test still passes. CI will check this for us.
https://github.com/ZcashFoundation/zebra/blob/dc9da24c8b6cee5c05315f0d77bee52613f75fd3/zebra-state/src/service/finalized_state.rs#L298-L311

Reply:
We should at least drop the _guard in the in the tracing component's `before_shutdown()` before exiting so lines in the buffer are flushed